[zhengsterz] iP

zhengsterz opened the pull request:
# RED : Robotic Enhancer of Development
>  “Everybody wants to be a bodybuilder, but nobody wants to lift no heavy-ass weights.” ― Ronnie Coleman([source](https://www.youtube.com/watch?v=5i5jGU9wn2M))

RED frees your mind of having to remember things you need to do. It's,
- text-based
- text-based
- ~~FAST~~ **SUPER** FAST to use

All you need to do is,
1. Download it from [here](https://github.com/zhengsterz/ip/releases/tag/untagged-ffdf0c57b374fbd7d7d1)
2. Double-click it
3. Add your tasks
4. Let it manage tasks for you 👍 

And it is **FREE!!!!**

Features :
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

Commit:
Add Gradle support

zhengsterz committed:
Level-1: Greet, Echo, Exit

zhengsterz committed:
Level 2. Add, List(Added a TaskList class)

zhengsterz committed:
Level-3: Mark as Done(Added Task Class)

zhengsterz committed:
Level-4: ToDo, Event, Deadline

zhengsterz committed:
A-TextUiTesting: Automated Text UI Testing

zhengsterz committed:
Level-5: Handle Errors

zhengsterz committed:
Level-6: Delete

zhengsterz committed:
README update

zhengsterz committed:
Add features that save the tasks in the hard disk automatically
whenever the task list changes

TaskFile class added to maintain tasklist.txt

Updates to Red and Tasklist classes to enable this new feature.

zhengsterz committed:
Add feature for DeadlineTask objects to store dates and time

zhengsterz committed:
Add ability to load tasks from harddrive back into list.

zhengsterz committed:
Merge branch 'branch-Level-7'

zhengsterz committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/DeadlineTask.java
#	src/main/java/Red.java

zhengsterz committed:
Add other classes to achieve better OOP.

Command and Parser classes added,TaskFile was changed to Storage
for better clarity and updates to all Task classes to
accomodate these changes

zhengsterz committed:
Remove foul language used in text

Classes were already put into packages in previous commit

zhengsterz committed:
Merge branch 'add-gradle-support'

zhengsterz committed:
Add JUnit tests for TodoTask and DeadlineTask

zhengsterz committed:
Add minor changes to make new branch for jar component.

zhengsterz committed:
Add some headers for Command classes

zhengsterz committed:
Add header comments to majority of the classes/methods.

zhengsterz committed:
Alter code to comply with coding standard

zhengsterz committed:
Add function to find specific tasks via keywords.

FindCommand added to facilitate this.

zhengsterz committed:
Merge branch 'branch-A-JavaDoc'

zhengsterz committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/red/storage/Storage.java

zhengsterz committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/red/parser/Parser.java
#	src/main/java/red/task/TaskList.java

JunHao247 commented on data/tasklist.txt:
```diff
@@ -0,0 +1,7 @@
+This is your current task list:
+1. [E][ ] cry         (From: 2pm        To: 8am) 
```
Maybe you could include the date along with the time? Since in your code  for deadline and event you are using LocalDateTime

JunHao247 commented on src/main/java/red/storage/Storage.java:
```diff
@@ -0,0 +1,136 @@
+import red.task.ToDoTask;
+
+import red.ui.UI;
+
```
I like how the imports are separated with a spacing into each category, very organised and tidy to read

JunHao247 commented on src/main/java/red/task/EventTask.java:
```diff
@@ -0,0 +1,28 @@
+public class EventTask extends Task {
+
+    protected String by;
+    protected String byy;
```
Would help to have a more detailed attribute like 
```
protected String start;
protected String end;
```

JunHao247 commented on src/main/java/red/storage/Storage.java:
```diff
@@ -0,0 +1,136 @@
+                    String[] arrOfStrStr= arrOfStr[1].split("\\(From: ", 2);
+                    String[] arrOfStrStrStr= arrOfStrStr[1].split("To: ", 2);
+                    String[] arrOfStrStrStrStr= arrOfStrStrStr[1].split("\\)", 2);
+                    EventTask currentTask = new EventTask(arrOfStrStr[0],arrOfStrStrStr[0],arrOfStrStrStrStr[0]);
```
It will be good to change the variable name to something more specific,
for example 
```
String[] arrOfFrom= arrOfStr[1].split("\\(From: ", 2);
```

similar for the other variable names

JunHao247 reviewed (commented):
Looks good, a few cosmetic changes requested, coding standards can be improved by changing some of the variable names and attribute names, I like the quote in the description 👍

zhengsterz committed:
Remove check statement

zhengsterz committed:
add gui capabilities

zhengsterz committed:
Merge branch 'branch-Level-10'

zhengsterz committed:
Add ability to run shadowJar by altering build gradle plus minor
commits

zhengsterz committed:
Merge pull request #1 from zhengsterz/branch-A-Assertions

Add ability to run shadowJar by altering build gradle plus minor

zhengsterz committed:
Add assertions to Un/MarkCommand,Parser,DeadlineTask and
CommandTask classes

Un/MarkCommand assertions checks that the tasklist size will never
be a negative number.Parser assertions checks that the parser
does not read an empty line as user input.DeadlineTask assertions
checks that the date and time provided conform to the necessary
format.

These ensure that if something were to go wrong it would be the
result of the user input instead of a problem with the program
itself.

zhengsterz committed:
Merge pull request #2 from zhengsterz/branch-A-Assertions

Add assertions to Un/MarkCommand,Parser,DeadlineTask and

zhengsterz committed:
Revert "Add assertions to Un/MarkCommand,Parser,DeadlineTask and"

zhengsterz committed:
Revert "Add ability to run shadowJar by altering build gradle plus minor"

zhengsterz committed:
Merge pull request #3 from zhengsterz/revert-1-branch-A-Assertions

Revert "Add ability to run shadowJar by altering build gradle plus mi…

zhengsterz committed:
Merge pull request #4 from zhengsterz/revert-2-branch-A-Assertions

Revert "Add assertions to Un/MarkCommand,Parser,DeadlineTask and"

zhengsterz committed:
Revert "Revert "Add ability to run shadowJar by altering build gradle plus mi…"

zhengsterz committed:
Merge pull request #5 from zhengsterz/revert-3-revert-1-branch-A-Assertions

Revert "Revert "Add ability to run shadowJar by altering build gradle plus mi…"

zhengsterz committed:
Revert "Revert "Add assertions to Un/MarkCommand,Parser,DeadlineTask and""

zhengsterz committed:
Merge pull request #6 from zhengsterz/revert-4-revert-2-branch-A-Assertions

Revert "Revert "Add assertions to Un/MarkCommand,Parser,DeadlineTask and""

zhengsterz committed:
Add booleans to some of the command classes to simplify
expressions

This was done to avoid complicated expressions, especially those
having many negations and nested parentheses, thus increasing
code quality.

zhengsterz committed:
Merge pull request #7 from zhengsterz/branch-A-CodeQuality

Add booleans to some of the command classes to simplify

zhengsterz committed:
Minor commmits

zhengsterz committed:
Add feature that prevents identical tasks from being added

zhengsterz committed:
Debug detecting duplicates feature

zhengsterz committed:
Change UI picture

zhengsterz committed:
Update user guide

zhengsterz committed:
ui png change

zhengsterz committed:
test

zhengsterz committed:
ui png update

zhengsterz committed:
test

zhengsterz committed:
user guide format update

zhengsterz committed:
user guide format update

zhengsterz committed:
Simplify Deadline and Event tasks
